Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -338,48 +387,84 @@ -(void)updateDisplayOrientedPose:(ARCamera*)camera {
 }
 
 - (void)session:(ARSession *)__unused session didAddAnchors:(nonnull NSArray<__kindof ARAnchor *> *)anchors {
-    if (!planeDetectionEnabled) {
-        return;
+    if (planeDetectionEnabled) {
+        [self LockPlanes];
+        for (ARAnchor* newAnchor : anchors) {
+            if ([newAnchor isKindOfClass:[ARPlaneAnchor class]]) {
+                updatedPlanes.insert((ARPlaneAnchor*)newAnchor);
+            }
+        }
+
+        [self UnlockPlanes];
     }
 
-    [self LockPlanes];
-    for (ARAnchor* newAnchor : anchors) {
-        if ([newAnchor isKindOfClass:[ARPlaneAnchor class]]) {
-            updatedPlanes.insert((ARPlaneAnchor*)newAnchor);
+    if (@available(iOS 13.4, *)) {
+        if (meshDetectionEnabled) {
+            [self LockMeshes];
+            for (ARAnchor* newAnchor : anchors) {
+                if ([newAnchor isKindOfClass:[ARMeshAnchor class]]) {
+                    updatedMeshes.insert((ARMeshAnchor*)newAnchor);
+                }
+            }
         }
-    }
 
-    [self UnlockPlanes];
+        [self UnlockMeshes];
```

Comment:
I think the `[self UnlockMeshes]` is meant to be inside the `if (meshDetectionEnabled)` conditional block?